### ISSUE TITLE
NVDA 2021.3 Beta no longer announces backspaced characters in Visual Studio 2019

### ISSUE THREAD

**Issue by lukaszgo1** [nvaccess/nvda#13098]
### Steps to reproduce:
Open a file in Visual Studio, type some text and try to remove characters with backspace.
### Actual behavior:
Deleted characters are not announced.
### Expected behavior:
NVDA should announce deleted characters.
#### NVDA installed/portable/running from source:
Tried all 3
#### NVDA version:
NVDA 2021.3 beta 2, latest Alpha
#### Windows version:
Tested both on Windows 7 and Windows 11
#### Name and version of other software in use when reproducing the issue:
Visual Studio 2019
#### Other information about your system:
None relevant
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
The issue does not occur with 2021.2
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes
### Technical:
This is caused by PR #12868 - reverting it makes characters announced again. It looks like  Visual Studio does not emit UIA  caret events and this just worked previously by chance since the MSAA caret events were routed to the focused object even though it was UIA.
The best solution for now I believe would be to restrict fix from #12868 to the MS Word controls, as we are not aware about any other problematic places.

@michaelDCurran  Thoughts?
